### PR TITLE
Plugins: Ensure `fetch` loading strategy cases for nested plugins

### PR DESCRIPTION
**Pull request by wbrowne**
**Why do we need this feature?**

This PR fixes an issue where child plugins will be given the `script` loading strategy in cases where they should be `fetch`

### CODE REVIEW

**Review by xnyo — APPROVED** [review]
LGTM! I had to read the logic multiple times as it was not super clear :sweat_smile: But that's how it works and couldn't figure out an easier way to express it :+1:

File: pkg/services/pluginsintegration/pluginassets/pluginassets_test.go [hunk 1]
```diff
@@ -76,6 +76,43 @@ func TestService_Calculate(t *testing.T) {
 			}),
 			expected: plugins.LoadingStrategyScript,
 		},
+		{
```
Comment (jackw) [inline pkg/services/pluginsintegration/pluginassets/pluginassets_test.go:79]: 👏 Lovely you added the tests given this logic isn't so straight forward.

**Review by jackw — APPROVED** [review]
LGTM! 🚀

File: pkg/services/pluginsintegration/pluginassets/pluginassets.go [hunk 2]
```diff
@@ -76,6 +83,6 @@ func (s *Service) compatibleCreatePluginVersion(ps map[string]string) bool {
 	return false
 }
 
-func (s *Service) cndEnabled(p pluginstore.Plugin) bool {
-	return s.cdn.PluginSupported(p.ID) || p.Class == plugins.ClassCDN
+func (s *Service) cndEnabled(pluginID string, class plugins.Class) bool {
```
Comment (marefr) [inline pkg/services/pluginsintegration/pluginassets/pluginassets.go:86]:
```suggestion
func (s *Service) cdnEnabled(pluginID string, class plugins.Class) bool {
```

File: pkg/services/pluginsintegration/pluginassets/pluginassets.go [hunk 3]
```diff
@@ -46,16 +46,23 @@ func (s *Service) LoadingStrategy(_ context.Context, p pluginstore.Plugin) plugi
...
+
+		// Since the parent plugin is not explicitly configured as script loading compatible,
+		// If the plugin is either loaded from the CDN (via its parent) or contains Angular, we should use fetch
+		if s.cndEnabled(p.Parent.ID, p.Class) || p.Angular.Detected {
+			return plugins.LoadingStrategyFetch
+		}
 	}
 
-	if !s.cndEnabled(p) && !p.Angular.Detected {
+	if !s.cndEnabled(p.ID, p.Class) && !p.Angular.Detected {
```
Comment (marefr) [inline pkg/services/pluginsintegration/pluginassets/pluginassets.go:65]:
```suggestion
	if !s.cdnEnabled(p.ID, p.Class) && !p.Angular.Detected {
```

File: pkg/services/pluginsintegration/pluginassets/pluginassets.go [hunk 4]
```diff
@@ -46,16 +46,23 @@ func (s *Service) LoadingStrategy(_ context.Context, p pluginstore.Plugin) plugi
...
+		// Check the parent's create_plugin_version setting
 		if pCfg, ok := s.cfg.PluginSettings[p.Parent.ID]; ok {
 			if s.compatibleCreatePluginVersion(pCfg) {
 				return plugins.LoadingStrategyScript
 			}
 		}
+
+		// Since the parent plugin is not explicitly configured as script loading compatible,
+		// If the plugin is either loaded from the CDN (via its parent) or contains Angular, we should use fetch
+		if s.cndEnabled(p.Parent.ID, p.Class) || p.Angular.Detected {
```
Comment (marefr) [inline pkg/services/pluginsintegration/pluginassets/pluginassets.go:60]:
```suggestion
		if s.cdnEnabled(p.Parent.ID, p.Class) || p.Angular.Detected {
```

**Review by marefr — APPROVED** [review]
LGTM with some minor suggestions

File: pkg/services/pluginsintegration/pluginassets/pluginassets.go (hunk 2 above)
Comment (wbrowne) [reply pkg/services/pluginsintegration/pluginassets/pluginassets.go]: Good spot!!

**Pull request merged**